src/hist.jl

```diff
@@ -191,6 +191,15 @@ mutable struct Histogram{T<:Real,N,E} <: AbstractHistogram{T,N,E}
         closed == :right || closed == :left || error("closed must :left or :right")
         isdensity && !(T <: AbstractFloat) && error("Density histogram must have float-type weights")
         _edges_nbins(edges) == size(weights) || error("Histogram edge vectors must be 1 longer than corresponding weight dimensions")
+        # We do not handle -0.0 in ranges correctly in `binindex` for performance
+        # Constructing ranges starting or ending with -0.0 is very hard,
+        # and ranges containing -0.0 elsewhere virtually impossible,
+        # so check the former just in case as it is cheap
+        foreach(edges) do e
+            e isa AbstractRange &&
+                (isequal(first(e), -0.0) || isequal(last(e), -0.0)) &&
```

Comment:
why not `any(isequal(-0.0), e)` as this would be a bit safer and cost is negligible I think?

Reply:
Yeah, I guess in normal use the number of bins is much lower than the number of observations so checking for all of them has a negligible cost. I've pushed a commit to do that.